openfe/protocols/restraint_utils/openmm/omm_restraints.py

```diff
@@ -0,0 +1,703 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+"""
+Classes for applying restraints to OpenMM Systems.
+
+Acknowledgements
+----------------
+Many of the classes here are at least in part inspired from
+`Yank <https://github.com/choderalab/yank/>`_ and
+`OpenMMTools <https://github.com/choderalab/openmmtools>`_.
+
+TODO
+----
+* Add relevant duecredit entries.
+* Add Periodic Torsion Boresch class
+"""
+import abc
+
+import numpy as np
+import openmm
+from openmm import unit as omm_unit
+from openmmtools.forces import (
+    HarmonicRestraintForce,
+    HarmonicRestraintBondForce,
+    FlatBottomRestraintForce,
+    FlatBottomRestraintBondForce,
+)
+from openmmtools.states import GlobalParameterState, ThermodynamicState
+from openff.units.openmm import to_openmm, from_openmm
+from openff.units import unit
+
+from gufe.settings.models import SettingsBaseModel
+
+from openfe.protocols.restraint_utils.geometry import (
+    BaseRestraintGeometry,
+    DistanceRestraintGeometry,
+    BoreschRestraintGeometry
+)
+
+from openfe.protocols.restraint_utils.settings import (
+    DistanceRestraintSettings,
+    BoreschRestraintSettings,
+)
+
+from .omm_forces import (
+    get_custom_compound_bond_force,
+    add_force_in_separate_group,
+    get_boresch_energy_function,
+)
+
+
+class RestraintParameterState(GlobalParameterState):
+    """
+    Composable state to control `lambda_restraints` OpenMM Force parameters.
+
+    See :class:`openmmtools.states.GlobalParameterState` for more details.
+
+    Parameters
+    ----------
+    parameters_name_suffix : Optional[str]
+      If specified, the state will control a modified version of the parameter
+      ``lambda_restraints_{parameters_name_suffix}` instead of just
+      ``lambda_restraints``.
+    lambda_restraints : Optional[float]
+      The strength of the restraint. If defined, must be between 0 and 1.
+
+    Acknowledgement
+    ---------------
+    Partially reproduced from Yank.
+    """
+
+    lambda_restraints = GlobalParameterState.GlobalParameter(
```

Comment:
Does ``1`` here mean that the restraint is fully turned on? Maybe add a comment here?

Reply:
Took me a while to remember, the answer is yes - assuming the underlying Force does it properly ;)

What I mean here is that technically it's up to the underlying for to define how the controlling parameter is applied, so in all the cases below it's "lambda_restraints * E" but nothing really stops it from being "1/lambda_restraints * E" is someone wanted to cause chaos.

That being said, I'm pretty sure we'd do our due dilligence to not let that happen..

This is all to say "Yes I've added the comment and updated the docstring" :)